<h1>NGCircos.r</h1>

<p>A R package for Next Generation Circos(NGCircos)</p>

<h2>Introduction</h2>

<p>NGCircos is developed in JavaScripts based on D3.js and JQuery.js</p>

<p>NGCircos is originally published at NAR Genomics and Bioinformatics and available at https://wlcb.oit.uci.edu/NG-Circos </p>

<p>Since R language is much more friendly in data visualization and plot programming than html, we ported NGCircos to this R package based on the <em>htmlwidgets</em> framework.</p>

<h2>Features</h2>

<ul>
<li>20 modules(<em>ARC</em>, <em>AUXILIARYLINE</em>, <em>BACKGROUND</em>, <em>BUBBLE</em>, <em>CHORD</em>, <em>CNV</em>, <em>COMBINATION</em>, <em>COMPARE</em>, <em>GENE</em>, <em>HEATMAP</em>, <em>HISTOGRAM</em>, <em>LEGEND</em>, <em>LINE</em>, <em>LINK</em>, <em>LOLLIPOP</em>, <em>REDIRECT</em>, <em>SCATTER</em>, <em>SNP</em>, <em>TEXT</em>, <em>WIG</em>) of NGCircos are ported to NGCircos.r and all parameters and animations are kept. All modules are now presented in function.</li>
<li>Instead of transforming data using python script in NGCircos, the data file can be directly input into NGCircos.r which will automatically transform data into data.frame and then be input to module function.</li>
<li>NGCircos.r is much more programming friendly than NGCircos. Users can draw a interactive Circos plot as easy as drawing a plot using ggplot2.</li>
</ul>

<h2>Installation</h2>

<pre><code>    # You need devtools for that
    if (!require('devtools')){install.packages('devtools')}
    devtools::install_github('mrcuizhe/NGCircos.r', build_vignettes = TRUE)
</code></pre>

<h2>Document</h2>

<p>Document is available at </p>

<ul>
<li><a href="https://github.com/mrcuizhe/NGCircos.r/blob/master/NGCircos.r_document.ipynb">NGCircos.r-jupyter notebook</a></li>
</ul>

<p>Or </p>

<ul>
<li><a href="https://github.com/mrcuizhe/NGCircos.r/blob/master/NGCircos_1.0.0.pdf">NGCircos.r-pdf</a></li>
</ul>

<h2>Contact</h2>

<p>Please contact cuizhe@hit.edu.cn or mrcuizhe@gmail.com for help</p>
