---
title: "NGCircos-r-vignette"
output: 
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 2
vignette: >
  %\VignetteIndexEntry{NGCircos-r-vignette}
  %\usepackage[utf8]{inputenc}
  %\VignetteEngine{knitr::rmarkdown}
date: "`r Sys.Date()`"
author: "Zhe Cui"
---

## Introduction

NGCircos is developed in Javascript and displayed in html, based on D3 and JQuery libraries. 
Since R language is much more friendly in data visualization and plot programming than html, we ported NGCircos to this R package based on the *htmlwidgets* framework.

## Features

* 20 modules(*ARC*, *AUXILIARYLINE*, *BACKGROUND*, *BUBBLE*, *CHORD*, *CNV*, *COMBINATION*, *COMPARE*, *GENE*, *HEATMAP*, *HISTOGRAM*, *LEGEND*, *LINE*, *LINK*, *LOLLIPOP*, *REDIRECT*, *SCATTER*, *SNP*, *TEXT*, *WIG*) of NGCircos are ported to NGCircos.r and all parameters and animations are kept. All modules are now presented in function.
* Instead of transforming data using python script in NGCircos, the data file can be directly input into NGCircos.r which will automatically transform data into data.frame and then be input to module function.
* NGCircos.r is much more programming friendly than NGCircos. Users can draw a interactive Circos plot as easy as drawing a plot using ggplot2.

## Installation

NGCircos.r is available at https://github.com/mrcuizhe/NGCircos.r

```{r eval=FALSE, screenshot.force=FALSE}
# You need devtools for that
if (!require('devtools')){install.packages('devtools')}

devtools::install_github('mrcuizhe/NGCircos.r', build_vignettes = TRUE)
```

## How to draw a interactive Circos plot using NGCircos.r

If you are a NGCircos user and familiar with the modules and its data structrue, it will only take you a few moment for getting used to NGCircos.r. 
Same as NGCircos, a main function called *NGCircos()* is neccesary to display the genome for Circos plot, which is the coordinates basis for all modules. 
20 Modules are added to genome through a parameter in *NGCircos()* called *tracklist*. An empty tracklist is set by default and module can be added by plusing a module to *tracklist*. For example, tracklist=NGCircosTextTrack()+NGCircosBackgroundTrack() means apply a TEXT module and BACKGROUND module together. The other parameters in *NGCircos()* are all correspond to that in Javascript version one to one.
For 3 of 20 modules(*COMBINATION*, *COMPARE*, *REDIRECT*), their parameters are included in *NGCircos()*
For the other 17 of 20 modules, which has a independent function(*NGCircosArcTrack()*, *NGCircosAuxLineTrack()*, *NGCircosBackgroundTrack()*, *NGCircosBubbleTrack()*, *NGCircosChordTrack()*, *NGCircosCnvTrack()*, *NGCircosGeneTrack()*, *NGCircosHeatmapTrack()*, *NGCircosHistogramTrack()*, *NGCircosLegendTrack()*, *NGCircosLineTrack()*, *NGCircosLinkTrack()*, *NGCircosLollipopTrack()*, *NGCircosScatterTrack()*, *NGCircosSnpTrack()*, *NGCircosTextTrack()*, *NGCircosWigTrack()*), all data(if yes) are input through a parameter called *data* in a data.frame structure. All other parameters are all correspond to that in Jacascript version one to one. 

### Genome configuration

default:

```{r plot1,fig.width=7, fig.height=6,echo=TRUE}
library(NGCircos)
p<-NGCircos()
```

The genome is set by defult as hg19, the genome can be changed by setting genome=list("Example1"=100,"Example2"=200,"Example3"=300). Meanwhile, the color of each chromosome can be changed by setting genomeFillColor = c("red","blue","green")

```{r plot2, fig.width=7, fig.height=6,echo=TRUE}
library(NGCircos)
p<-NGCircos(genome=list("Example1"=100,"Example2"=200,"Example3"=300),genomeFillColor = c("red","blue","green"))
```

The filling color for chromosome can also be set automatically by choosing a *RColorBrewer* palette, such as *Blues*
```{r plot3, fig.width=7, fig.height=6,echo=TRUE,eval=FALSE}
library(NGCircos)
NGCircos(genome=list("Example1"=100,"Example2"=200,"Example3"=300),genomeFillColor = "Blues")
```
 
Other parameters are documented and try ?NGCircos() for details
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
NGCircos(genome=list("Example1"=100,"Example2"=200,"Example3"=300),genomeFillColor = "Blues",chrPad = 0,genomeTicksDisplay = TRUE,genomeTicksScale = 10,genomeLabelDisplay = FALSE)
```

### Adding module through *tracklist*

17 modules(*NGCircosArcTrack()*, *NGCircosAuxLineTrack()*, *NGCircosBackgroundTrack()*, *NGCircosBubbleTrack()*, *NGCircosChordTrack()*, *NGCircosCnvTrack()*, *NGCircosGeneTrack()*, *NGCircosHeatmapTrack()*, *NGCircosHistogramTrack()*, *NGCircosLegendTrack()*, *NGCircosLineTrack()*, *NGCircosLinkTrack()*, *NGCircosLollipopTrack()*, *NGCircosScatterTrack()*, *NGCircosSnpTrack()*, *NGCircosTextTrack()*, *NGCircosWigTrack()*) are added by using "+" in *tracklist*, for example: 
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
tracklist<-NGCircosTextTrack('textTrack01', 'Annotation', color = '#DD2222', x = -40) + NGCircosBackgroundTrack('bgTrack01', fillColors="#FFEEEE", borderSize = 1)
NGCircos(tracklist = tracklist)
```

Details for each module are presented below:

## Auxiliary line module

There is no data input for Auxiliary line module. The details for parameters are available by ?NGCircosAuxLineTrack
An example for Auxiliary line module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
NGCircos(NGCircosAuxLineTrack('AuxLineTrack01'))
```

## Background module

There is no data input for Background module. The details for parameters are available by ?NGCircosBackgroundTrack
An example for Background module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos) 
NGCircos(NGCircosBackgroundTrack('bgTrack01', fillColors="#FFEEEE", borderSize = 1))
```

## Bubble module

The data is input throught the *data* parameter in bubble module. The details for parameters are available by ?NGCircosBubbleTrack
An example for bubble module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
bubbleData<-bubbleExample
NGCircos(NGCircosBubbleTrack('BubbleTrack01', maxRadius = 230, minRadius = 170, data=bubbleData, blockStroke = TRUE, bubbleMaxSize =10, bubbleMinSize = 2, maxColor = "red", minColor = "yellow", totalLayer =3, animationDisplay = TRUE, animationType="linear"),genome = list("2L"=23011544,"2R"=21146708,"3L"=24543557,"3R"= 27905053,"X"=22422827,"4"=1351857), BUBBLEMouseOverDisplay =TRUE,innerRadius = 236)
```

## Chord module

The data is input throught the *data* parameter in chord module. The details for parameters are available by ?NGCircosChordTrack
An example for chord module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
chordData<-chordExample
NGCircos(NGCircosChordTrack('CHORDTrack', data = chordData,innerRadius= 210,outerRadius= 211,fillOpacity=0.67,
strokeColor="black",strokeWidth= "1px",outerARCText=FALSE),genome=list("C.CK" = 189.51,"C.NPK"=188,"GC.CK"=186.11,
"GC.NPK"=191.51,"Alphaproteobacteria"=70.16,"Betaproteobacteria"=23.51,"Gammaproteobacteria"=25.51,
"Deltaproteobacteria"=23.28,"Acidobacteria"=53.62,"Actinobacteria"=72.33,"Bacteroidetes"=22.41,
"Chloroflexi"=15.08,"Firmicutes"=10.72,"Gemmatimonadetes"=26.37,"Planctomycetes"=19.26,"Thaumarchaeota"=6.15,
"Verrucomicrobia"=8.3,"Ascomycota"=159.41,"Basidiomycota"=79.73,"Zygomycota"=139.29 ),outerRadius = 217,
genomeLabelDisplay = FALSE)
```

## Cnv module

The data is input throught the *data* parameter in cnv module. The details for parameters are available by ?NGCircosCnvTrack
An example for cnv module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
cnvData<-cnvExample
NGCircos(NGCircosCnvTrack('CnvTrack01',maxRadius =175, minRadius =116, data =cnvData,width=2,color = "#4876FF")+
NGCircosBackgroundTrack("bgTrack01",minRadius = 116,maxRadius = 175,fillColors = "#F2F2F2",axisShow = TRUE),CNVMouseOverDisplay = TRUE)
```

## Gene module

The data is input throught the *data* parameter in gene module. The details for parameters are available by ?NGCircosGeneTrack
An example for gene module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
geneData<-geneExample
NGCircos(NGCircosGeneTrack('GeneTrack01', outerRadius = 195, innerRadius = 180, data=geneData,arrowGap = 10,
 arrowColor = "black",arrowSize = "12px",cdsColor = "#1e77b3",cdsStrokeColor = "#1e77b3",cdsStrokeWidth= 5,
 utrWidth= -2,utrColor= "#fe7f0e",utrStrokeColor= "#fe7f0e",animationDisplay = TRUE),genome =list("EGFR"=1000),
 outerRadius = 220)
```

## Heatmap module

The data is input throught the *data* parameter in heatmap module. The details for parameters are available by ?NGCircosHeatmapTrack
An example for heatmap module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
heatmapData<-heatmapExample
NGCircos(NGCircosHeatmapTrack('HeatmapTrack01', maxRadius= 180, minRadius = 100, data=heatmapData,totalLayer = 3),
genome = list("2L"=23011544,"2R"=21146708,"3L"=24543557,"3R"=27905053,"4"=1351857,"X"=22422827),
HEATMAPMouseEvent = TRUE,HEATMAPMouseOverDisplay = TRUE)
```

## Histogram module

The data is input throught the *data* parameter in histogram module. The details for parameters are available by ?NGCircosHistogramTrack
An example for histogram module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
histogramData<-histogramExample
NGCircos(NGCircosHistogramTrack('HISTOGRAMTrack', data = histogramData,fillColor= "#ff7f0e",maxRadius = 210,
minRadius = 175),genome=list("2L"=23011544,"2R"=21146708,"3L"=24543557,"3R"= 27905053,"X"=22422827,"4"=1351857),
outerRadius = 220)
```

## Legend module

There is no data input for legend module. The details for parameters are available by ?NGCircosLegendTrack
An example for legend module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
legend1 <- list(type= "circle", color="#1E77B4",opacity="1.0",circleSize="8",text= "C.CK", textSize= "14",textWeight="normal")
legend2 <- list(type= "circle", color="#AEC7E8",opacity="1.0",circleSize="8",text= "C.NPK", textSize= "14",textWeight="normal")
NGCircos(NGCircosLegendTrack('legendTrack01', title = "legend",data=list(legend1,legend2),size = 20))
```

## Line module

The data is input throught the *data* parameter in line module. The details for parameters are available by ?NGCircosLineTrack
An example for line module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
lineData<-lineExample
NGCircos(NGCircosLineTrack('LINETrack', data = lineData,maxRadius=200,minRadius=150,color= "#ff0031")+
NGCircosBackgroundTrack('BGTrack',minRadius = 205,maxRadius = 150))
```

## Link module

The data is input throught the *data* parameter in link module. The details for parameters are available by ?NGCircosLinkTrack
An example for link module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
linkData<-linkExample
NGCircos(NGCircosLinkTrack('LINKTrack', data = linkData,LinkRadius= 140,fillColor= "#9e9ac6",width= 2,
axisPad= 3,labelPad=8,animationDisplay=TRUE,animationDirection="1to2", animationType= "linear" ))
```

## Lollipop module

The data is input throught the *data* parameter in lollipop module. The details for parameters are available by ?NGCircosLollipopTrack
An example for lollipop module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
lollipopData<-lollipopExample
arcData<-arcExample
NGCircos(NGCircosLollipopTrack('LollipopTrack01', data=lollipopData, fillColor="#9400D3",
circleSize= 6, strokeColor= "#999999", strokeWidth= "1px", animationDisplay=TRUE, lineWidth= 2,
realStart= 101219350)+NGCircosArcTrack('ArcTrack01', outerRadius = 212, innerRadius = 224, data=arcData),
 genome=list("EGFR"=1211),outerRadius = 220,genomeFillColor = c("grey"))
```

## Snp module

The data is input throught the *data* parameter in snp module. The details for parameters are available by ?NGCircosSnpTrack
An example for snp module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
snpData<-snpExample
NGCircos(NGCircosSnpTrack('SNPTrack', minRadius =150, maxRadius = 190, data = snpExample,SNPFillColor= "#9ACD32", circleSize= 2, SNPAxisColor= "#B8B8B8", SNPAxisWidth= 0.5, SNPAnimationDisplay=TRUE,SNPAnimationTime= 2000, SNPAnimationDelay= 0, SNPAnimationType= "linear") + NGCircosBackgroundTrack('BGTrack',minRadius = 145,maxRadius = 200))
```

## Text module

There is no data input for text module. The details for parameters are available by ?NGCircosTextTrack
An example for text module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)  
NGCircos(NGCircosTextTrack('textTrack01', 'Annotation', color = '#DD2222', x = -40))
```

## Wig module

The data is input throught the *data* parameter in wig module. The details for parameters are available by ?NGCircosWigTrack
An example for wig module:
```{r, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
wigData<-wigExample
NGCircos(NGCircosWigTrack('WIGTrack', data = wigData, maxRadius= 200,minRadius= 150,strokeColor= "darkblue",
color= "lightblue",strokeType= "cardinal")+NGCircosBackgroundTrack('BGTrack',minRadius = 205,maxRadius = 150)
,genome=list("chr8"=1000),outerRadius = 220)
```

## Saving plot

If you are using RStudio, the plot can be saved as an image or a html page.
If you are using R in command, the plot can be saved as a html page, thanks to the *htmlwidgets* library:
```{r eval=FALSE, screenshot.force = FALSE,eval=FALSE}
library(NGCircos)
library(htmlwidgets)

wigData<-wigExample
plot <- NGCircos(NGCircosWigTrack('WIGTrack', data = wigData, maxRadius= 200,minRadius= 150,strokeColor= "darkblue",
color= "lightblue",strokeType= "cardinal")+NGCircosBackgroundTrack('BGTrack',minRadius = 205,maxRadius = 150)
,genome=list("chr8"=1000),outerRadius = 220)

saveWidget(plot, "plot.html")
```

## Citation

The NGCircos is originally published at 
The NGCircos.r is originally published at 

Please contact cuizhe@hit.edu.cn or mrcuizhe@gmail.com for any question
